### PR TITLE
A new tank with arc-track bullets

### PR DESCRIPTION
**Pull request by TLLtangling**
just a useless test

### CODE REVIEW

File: src/battle_game/core/game_core.h [hunk 1]
```diff
@@ -117,6 +118,7 @@ class GameCore {
   void SetRenderPerspective(uint32_t player_id);
   [[nodiscard]] uint32_t GetRenderPerspective() const;
   [[nodiscard]] glm::vec4 GetPlayerColor(uint32_t player_id) const;
+  [[nodiscard]] glm::vec4 GetPlayerColorbooboo(uint32_t player_id) const;
```
Comment (LazyJazz) [inline src/battle_game/core/game_core.h:121]: What is this for? If this is not generally used, pls do this in your local codes. you can use GetRenderPerspective() to get the internal value render_perspective_

File: src/battle_game/core/units/boobootank.h [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#pragma once
+#include "battle_game/core/units/tiny_tank.h"
+#include "battle_game/core/unit.h"
+
+namespace battle_game::unit {
+class boobootank : public Tank {
```
Comment (LazyJazz) [inline src/battle_game/core/units/boobootank.h:6]: BigCamelCase

File: src/battle_game/core/units/boobootank.cpp [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+  if (player) {
+    auto &input_data = player->GetInputData();
+    auto diff = input_data.mouse_cursor_position - position_;
+    if (glm::length(diff) < 1e-4) {
+      turret_rotation_ = rotation_;
+    } else {
+      turret_rotation_ = std::atan2(diff.y, diff.x) - glm::radians(90.0f);
+    }
+  }
+}
```
Comment (LazyJazz) [inline src/battle_game/core/units/boobootank.cpp:131]: If you did not make changes to these functions, inherit from unit::Tank.

File: src/battle_game/core/bullets/booboobullet2.h [hunk 4]
```diff
@@ -0,0 +1,23 @@
+#pragma once
+#include "battle_game/core/bullet.h"
+
+namespace battle_game::bullet {
+class booboobullet2 : public Bullet {
```
Comment (LazyJazz) [inline src/battle_game/core/bullets/booboobullet2.h:5]: BigCamelCase

File: src/battle_game/core/bullets/booboobullet.h [hunk 5]
```diff
@@ -0,0 +1,23 @@
+#pragma once
+#include "battle_game/core/bullet.h"
+
+namespace battle_game::bullet {
+class booboobullet : public Bullet {
```
Comment (LazyJazz) [inline src/battle_game/core/bullets/booboobullet.h:5]: BigCamelCase

**Review by LazyJazz — CHANGES_REQUESTED** [review]

File: src/battle_game/core/units/BoobooTank.cpp [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+  battle_game::DrawModel(tank_body_model_index);
+  battle_game::SetRotation(turret_rotation_);
+  battle_game::DrawModel(tank_turret_model_index);
+}
+
+void BoobooTank::Update() {
+  TankMove(3.0f, glm::radians(180.0f));
+  TurretRotate();
+  Fire();
+}
```
Comment (qinchuanhui) [inline src/battle_game/core/units/BoobooTank.cpp:86]: You should not rewrite this function again.

File: src/battle_game/core/units/BoobooTank.cpp [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+        turret_vertices.push_back({{sin_theta * 0.5f, cos_theta * 0.5f},
+                                   {0.0f, 0.0f},
+                                   {0.7f, 0.7f, 0.7f, 1.0f}});
+        turret_indices.push_back(i);
+        turret_indices.push_back((i + 1) % precision);
+        turret_indices.push_back(precision);
+      }
+      turret_vertices.push_back(
+          {{0.0f, 0.0f}, {0.0f, 0.0f}, {0.7f, 0.7f, 0.7f, 1.0f}});
+      turret_vertices.push_back(
```
Comment (qinchuanhui) [inline src/battle_game/core/units/BoobooTank.cpp:53]: Why rewrite this code, it seems to do the same thing as constructor-function of class `Tank`.
You can look at codes of other, if you don't know how to inherit the Tank class correctly.

**Review by qinchuanhui — CHANGES_REQUESTED** [review]

**Review by qinchuanhui — APPROVED** [review]
OK, you can get the corresponding score

**Pull request closed without merging**